src/network/connection.cpp

```diff
@@ -249,15 +246,17 @@ BufferedPacket ReliablePacketBuffer::popSeqnum(u16 seqnum)
 	}
 
 	m_list.erase(r);
-	--m_list_size;
 
-	if (m_list_size == 0)
-	{ m_oldest_non_answered_ack = 0; }
-	else
-	{ m_oldest_non_answered_ack = readU16(&(*m_list.begin()).data[BASE_HEADER_SIZE+1]);	}
+	if (m_list.empty()) {
+		m_oldest_non_answered_ack = 0;
+	} else {
+		m_oldest_non_answered_ack =
+				readU16(&(*m_list.begin()).data[BASE_HEADER_SIZE+1]);
```

Comment:
Dereferencing could be replaced with 'pointer call' notation (or what the term for this is).
Same for `ReliablePacketBuffer::popFirst()`
Also: Missing spaces around `+`